md/connector-archetype.md

```diff
@@ -0,0 +1,273 @@
+    }
+    
+    def checkMandatoryStringInput(inputName) throws ConnectorValidationException {
+        def value = getInputParameter(inputName)
+        if (value in String) {
+            if (!value) {
+                throw new ConnectorValidationException(this, "Mandatory parameter '$inputName' is missing.")
+            }
+        } else {
+            throw new ConnectorValidationException(this, "'$inputName' parameter must be a String")
+        }
+    }
+    
+    /**
+     * Core method:
+     * - Execute all the business logic of your connector using the inputs (connect to an external service, compute some values ...).
+     * - Set the output of the connector execution. If outputs are not set, connector fails.
+     */
+    @Override
+    def void executeBusinessLogic() throws ConnectorException {
+        def defaultInput = getInputParameter(defaultInput)
+        setOutputParameter(defaultOutput, "$defaultInput - output".toString())
+    }
+    
+    /**
+     * [Optional] Open a connection to remote server
+     */
+    @Override
+    def void connect() throws ConnectorException{}
+
+    /**
+     * [Optional] Close connection to remote server
+     */
+    @Override
+    def void disconnect() throws ConnectorException{}
+}
+```
+
+The methods _validateInputParameters_ and _executeBusinessLogic_ must be implemented, and are called by the Bonita engine when the connector is executed.  
+The methods _connect_ and _disconnect_ can be used to open and close a connection to a remote server.  The life cycle of the connection will then be managed by the Bonita engine.
+
+#### Build a connector project
+
+A connector project is built using Maven, and especially the [maven assembly plugin](https://maven.apache.org/plugins/maven-assembly-plugin/).   
+The root _pom.xml_ file has the following parent: 
+```xml
+<parent>
+    <groupId>org.bonitasoft.connectors</groupId>
+    <artifactId>bonita-connectors</artifactId>
+    <version>1.0.0</version>
+</parent>
+```
+This parent contains the logic that make the replacements in the implementation xml file at build time.
+
+By default, two zip archives are built: 
+
+ - One containing all the definitions and implementations found in the project (built using the file _connector-assembly.xml)_
+ - One containing only the default implementation generated (built using the file _[connector name]-assembly.xml_
+
+Those two assembly are here to help you to:
+
+ - Build an _all in one_ zip archive for all the definitions and implementation created in this project. By importing this archive in a Bonita Studio you will import all the definitions and implementations created in the project
+ - Build a single zip archive with only one implementation. This implementation zip archive can be imported in a Bonita Studio, but also in a Bonita Bundle to update a connector implementation at runtime.
+
+To build the connector project, type the following command at the root of the project : 
+```
+mvn clean install
+```
```

Comment:
```suggestion
```
By default, maven wrapper is installed by the archetype. It helps to have a portable and reproductible build system. If you elected to disable the wrapper install at archetype generation, you can use your local maven install:
```
mvn clean install
```
```